.github/workflows/build.yml

```
@@ -35,6 +35,17 @@ jobs:
       - name: Run unit tests
         run: make test
       - name: Run e2e tests
-        run:  make e2e
+        run:  |
+          #!/bin/sh
+          if [[ $GITHUB_REF_NAME == v* && $GITHUB_REF_TYPE == tag ]]; then
+            make e2e
+          else
+            export GOCOVERDIR=$GITHUB_WORKSPACE/test/e2e/.cover
+            rm $GOCOVERDIR -rf
+            mkdir $GOCOVERDIR -p
```

nit: write options before the arguments for portability. 
```suggestion
            rm -rf $GOCOVERDIR
            mkdir -p $GOCOVERDIR
```